// Copyright 2017 gRPC authors.

syntax = "proto3";

import "tests/testing/proto/requests.proto";

package tests_of_grpc_testing;

message Down {
  int32 first_down_field = 1;
}

message Strange {
  int32 first_strange_field = 1;
}

message Bottom {
  int32 first_bottom_field = 1;
}

service FirstService {
  rpc UnUn(Up) returns (Down);
  rpc UnStre(Charm) returns (stream Strange);
  rpc StreUn(stream Charm) returns (Strange);
  rpc StreStre(stream Top) returns (stream Bottom);
}

service SecondService {
  rpc UnStre(Strange) returns (stream Charm);
}
